specification/dartLangSpec.tex

```diff
@@ -9807,23 +9796,20 @@ \subsubsection{Cascades}
 }
 
 \LMHash{}%
-For some $k > 0$,
-let $e$ be an expression of the form
-\code{$e_0$?..$s_1$..$s_2$ $\cdots$ ..$s_k$}
-that occurs at some location $\ell$,
-such that $e_0$ is a \synt{conditionalExpression} and
-$s_j$ is a \synt{cascadeSection}, $j \in 1 .. k$.
-Assume that $e$ is maximal,
-in the sense that there does not exist a \synt{cascadeSection} $s$
-such that there is an expression at location $\ell$
-of the form \code{$e$..$s$}
-(\commentary{in short, we can not make $e$ bigger}).
-Evaluation of $e$ proceeds as follows:
+Evaluation of a cascaded member access $e$
+of the form \code{$e_0$..$s$} or the form \code{$e_0$?..$s$}
+where $e$ is an \synt{expression} and
+$s$ is a \synt{cascadeSection}
+proceeds as follows:
 Evaluate $e_0$ to an object $o$.
-If $o$ is the null object then $e$ evaluates to $o$.
-Otherwise, let \code{v} be a fresh variable bound to $o$.
-Then $e$ evaluates to the result of
-evaluating \code{v..$s_1$..$s_2$ $\cdots$ ..$s_k$}.
+If $o$ is not the null object:
+let \code{v} be a fresh variable bound to $o$ and
+evaluate \code{v.$s$} to an object which is discarded,
+then $e$ evaluates to $o$.
+Otherwise $o$ is the null object.
+If $e$ is initially conditional then $e$ evaluates to $o$;
+%% TODO(eernst): With NNBD, this error cannot happen -- delete.
+otherwise a dynamic error occurs.
```

Comment:
How about:

> If $e$ is initially conditional and $o$ is the null object,
then $e$ evaluates to the null object.
Otherwise, let \code{v} be a fresh variable bound to $o$ and
evaluate \code{v.$s$} to an object which is discarded, and
then $e$ evaluates to $o$.

The currently proposed text incorrectly disallows `null..toString()`.

Reply:
Ah, yes, and the dynamic error will happen during `v.s` anyway. Done.